src/emulator.ts

```diff
@@ -541,4 +564,61 @@ export class EmacsEmulator implements IEmacsController, vscode.Disposable {
       }
     });
   }
+
+  public saveRegister(arg: string): void {
+    const selectionsAfterRectDisabled =
+      this.inRectMarkMode &&
+      this.nativeSelections.map((selection) => {
+        const newLine = selection.active.line;
+        const newChar = Math.min(selection.active.character, selection.anchor.character);
+        return new vscode.Selection(newLine, newChar, newLine, newChar);
+      });
+    const selections = getNonEmptySelections(this.textEditor);
+    if (selectionsAfterRectDisabled) {
+      this.textEditor.selections = selectionsAfterRectDisabled;
+    }
+    // selections is now a list of non empty selections, iterate through them and
+    // build a single variable combinedtext
+    let i = 0;
+    let combinedtext = "";
+    while (i < selections.length) {
+      combinedtext = combinedtext + this.textEditor.document.getText(selections[i]);
```

Comment:
While it's not needed in this PR, the copy/paste functions of the kill/yank commands  should be shared with these methods. They are compatible with multi-cursors.

Reply:
A good point. Still researching